README.md

```diff
@@ -1,122 +1,237 @@
-
----
-### License
+<!--
+FUTURE: Add this paragraph:
 
-```
-   Copyright 2020 Andrew Chen Wang
+⚠️ Make sure that any changes made to the Markdown files in the GitHub Action
+_are also committed_! This GitHub Action uses `git subtree` which takes the
+state of the latest Git commit, not the current state of the working directory.
+You can use [EndBug/add-and-commit] or `git add -A` and
+`git commit --amend --no-edit` to commit changes inside your workflow.
 
-   Licensed under the Apache License, Version 2.0 (the "License");
-   you may not use this file except in compliance with the License.
-   You may obtain a copy of the License at
+-->
 
-       http://www.apache.org/licenses/LICENSE-2.0
+<!--
+FUTURE: Add a second example
 
-   Unless required by applicable law or agreed to in writing, software
-   distributed under the License is distributed on an "AS IS" BASIS,
-   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-   See the License for the specific language governing permissions and
-   limitations under the License.
+```yml
+name: Publish to GitHub wiki
+on:
+  push:
+    branches: [main]
+    paths: [wiki/**, .github/workflows/wiki.yml]
+concurrency:
+  group: wiki
+  cancel-in-progress: true
+permissions:
+  contents: write
+jobs:
+  wiki:
+    environment:
+      name: github-wiki
+      url: ${{ steps.github-wiki-action.outputs.page_url }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - run: date > my-wiki/Todays-date.md
+      - uses: EndBug/add-and-commit@v9
+      - uses: Andrew-Chen-Wang/github-wiki-action@v4
+        id: github-wiki-action
+        with:
+          path: my-wiki
 ```
+-->
+
+<!-- FUTURE: Change this to `path` instead of `WIKI_DIR` -->
+
+After creating your workflow file, now all you need is to put your Markdown
+files in a `wiki/` folder (or whatever you set the `WIKI_DIR` option to) and
+commit them to your default branch to trigger the workflow (or whatever other
+trigger you set up).
+
+💡 Each page has an auto-generated title. It is derived from the filename by
+replacing every `-` (dash) character with a space. Name your files accordingly.
+
+💡 The `Home.md` file will automatically become the homepage, not `README.md`.
+This is specific to GitHub wikis. You can use a preprocessor like [TBD] to
+automagically ✨ change `README.md` into `Home.md` and rewrite links from
+`./My-sibling-page.md` to `./My-sibling-page` to work when deployed to the
+GitHub wiki.
+
+### GitHub token
+
+This actions needs a GitHub token that can write to the GitHub wiki of the
+selected repository. By default, the `${{ github.token }}` you are given only
+offers **read** permissions to the current GitHub repository. You will need to
+either:
+
+1. Provide a custom GitHub PAT that has `contents: write` permissions. This can
+   be generated from [github.com/settings/personal-access-tokens].
+2. Upgrade the `${{ github.token }}` using the GitHub Actions' `permissions:`
```

Comment:
Please specify this is only possible on public repositories. I still need to see a repo where it's possible for private repos.

Reply:
Will do!

Reply:
Works in private repos
![image](https://user-images.githubusercontent.com/61068799/225440812-aebbf223-a304-4e17-9491-e7c737ef388c.png)
![image](https://user-images.githubusercontent.com/61068799/225440854-5f514740-e4f5-4481-8bde-87a85bf09e75.png)
```yml
name: Wiki
on:
  push:
    branches: [main]
    paths: [wiki/**, .github/workflows/publish-wiki.yml]
concurrency:
  group: wiki
  cancel-in-progress: true
permissions:
  contents: write
jobs:
  wiki:
    environment:
      name: github-wiki
      url: https://github.com/${{ github.repository }}/wiki
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - uses: Andrew-Chen-Wang/github-wiki-action@v3
        env:
          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          GH_MAIL: jcbhmr@outlook.com
          GH_NAME: jcbhmr
```

it should always be the same permissions and everything, private vs public doesn't matter. Just make sure you have GitHub Pro or EDU edition to enable private wikis!